diff --git a/supabase/_async/client.py b/supabase/_async/client.py
index 6992af0..398b7fb 100644
--- a/supabase/_async/client.py
+++ b/supabase/_async/client.py
@@ -65,10 +65,7 @@ class AsyncClient:
         self.supabase_url = supabase_url
         self.supabase_key = supabase_key
         self.options = copy.copy(options)
-        self.options.headers = {
-            **options.headers,
-            **self._get_auth_headers(),
-        }
+        self.options.headers = copy.copy(self._get_auth_headers())
 
         self.rest_url = f"{supabase_url}/rest/v1"
         self.realtime_url = f"{supabase_url}/realtime/v1".replace("http", "ws")
diff --git a/supabase/_sync/client.py b/supabase/_sync/client.py
index af2a841..f100b3b 100644
--- a/supabase/_sync/client.py
+++ b/supabase/_sync/client.py
@@ -64,10 +64,7 @@ class SyncClient:
         self.supabase_url = supabase_url
         self.supabase_key = supabase_key
         self.options = copy.copy(options)
-        self.options.headers = {
-            **options.headers,
-            **self._get_auth_headers(),
-        }
+        self.options.headers = copy.copy(self._get_auth_headers())
 
         self.rest_url = f"{supabase_url}/rest/v1"
         self.realtime_url = f"{supabase_url}/realtime/v1".replace("http", "ws")
